### PR TITLE
Add smooth-scroll for Scrolling Support for nav-bar Links

### LINKED ISSUE #14: Add Logic in NavBar for Scrolling to Relevant Sections
Something like the nav-bar at [https://service-link.ca/](something like the nav-bar at https://service-link.ca/). The method in which this is added should be discussed. 

Depends on #2

### PR DESCRIPTION
**Pull request by SudharakaP**
Add smooth-scroll library (https://github.com/cferdinandi/smooth-scroll) so that we have a very smooth scrolling experience when clicking on nav-bar links.

Fixes #14

### CODE REVIEW

File: src/components/layout.js [hunk 1]
```diff
@@ -13,6 +13,13 @@ import "./layout.css"
 import JumbotronElement from "./jumbotron"
 import Footer from "./footer";
 
+if (typeof window !== "undefined") {
```
Comment (faithghlee) [inline src/components/layout.js]: Actually i tried running with and without this section of code, and i dont observe any difference in terms of the scrolling effect. I am not entirely sure what is supposed to do for this

Comment (SudharakaP) [reply src/components/layout.js]: Can you do one thing; when you remove that part can you restart the server and start it again with `gatsby develop`? :smile:

Comment (SudharakaP) [reply src/components/layout.js]: Also do an `npm install` before the restart please since the `package.json` is updated in this pull request.

Comment (faithghlee) [reply src/components/layout.js]: works as intended, for some reason i was thinking the scroll bar was something else, didnt know its a redirection from nav.link to their respective pages. sorry for confusion

**Comment by SudharakaP** [comment]
@faithghlee : Hope we can make it look even more awesome with time :smile:

**Pull request merged**